### DISCUSSION TITLE
KiCad 7 new DNP flag

### DISCUSSION THREAD

**Discussion by set-soft** [INTI-CMNB/KiBot#395] [Ideas]
KiCad 7 introduced a new flag for the components named *Do not populate*.

This flag is independent of *Exclude from bill of materials* and *Exclude from board*. So is quite subtle, you can have a component in the BoM, with a footprint in the PCB, but just not soldered.

This is different to how KiBot uses DNP, because in KiBot you can choose to generate a separated BoM for DNPs.

This discussion is to propose ideas on how to handle this new flag in KiBot.

**Comment by mdeweerd** [comment]
I see this as another flag to define DNP.

We have an field/attribute "dnp" on a component that we use to indicate if the component has to be placed or not.  And that value can be changed by using the variants or configurations in KiBOT that (can) override this.

For now I think that this new flag can be "combined" with the dnp field.  If either the flag or the dnp field is "true", then the behavior is the same as the "dnp" field being set to true, and then all other current rules apply.

Two special cases that I think of:
- Test points - they are not mounted on the PCB as a component, but it is still usefull to list them in a position file;
- Connector-less programming pads - I use a Tag-connect programming cable that does not require a component on the boards, and I know there are other custom implementations of programming methods that basically use test-points and a custom cable to program.


With regards to "Exclude from board", I assume that this could still be an item that should be included in the BOM but that does not have a footprint - which could be a way to specify a fuse-holder and the fuse and the fuse-cap as two separate components.  One being on the board and the others having to be placed on the fuse-holder once it's soldered.

**Reply by set-soft** [reply]
The 6801dfc7e24138c6fc473f9af1393989048a16d9 patch copies the KiCad schematic flag to the internal *fitted* flag.

It also removes 3D models.

But I think we need more control over what is done by the filters
